src/lib/work_selector/offset.ml

```diff
@@ -0,0 +1,34 @@
+open Core_kernel
+
+module Make
+    (Inputs : Intf.Inputs_intf)
+    (Lib : Intf.Lib_intf with module Inputs := Inputs) =
+struct
+  (* Offset will be generated randomly at each execution, and will stay
+     constant for the duration of the execution *)
+  let offset =
+    (* Self init will make Random.int produce a new number at each execution *)
+    Random.self_init () ;
+    (* TODO what should be put as an upperbound for the offset? is it fine to not have perfectly uniform probability? *)
+    Random.int 1000
```

Comment:
As you are using `mod` later, maybe you can have here the highest possible bound (2^30 I believe).
That would be more uniform without much change.

Reply:
Yes, good idea, I’ll do that.

Reply:
Done.